data/transactions/logic/jsonspec_test.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/encoding/unicode/utf32"
+)
+
+// As of go1.17.5, json implements encoding and decoding of JSON as defined in RFC 7159. https://pkg.go.dev/encoding/json
+type utf8String string
```

Comment:
This concerns me.  Did Go *change* it's implementation and therefore before 1.17.5, we would get different behavior?  If so, we must either detect and fix, or demand compilation with 1.17.5 or greater, or else we'll get the potential for disagreement about results.

Reply:
I'll update this comment. Go json started using RFC 7159 in v1.10. I think we're fine since we are requiring build to use v1.16 or later.

Reply:
If we can find where they talk about this change, it might be a good source of ideas for tests.  They might describe small changes between the old and new implementation.

Reply:
https://datatracker.ietf.org/doc/html/rfc7159.html#section-1.3, it doesn't list behavior changes in this update.